query_server/query/src/extension/physical/transform_rule/table_writer.rs

```diff
@@ -69,7 +76,13 @@ impl ExtensionPlanner for TableWriterPlanner {
                     physical_inputs.len(),
                     "TableWriterMergePlanNode has multiple inputs."
                 );
-                let physical_input = physical_inputs[0].clone();
+
+                let batch_size = session_state.config().batch_size();
+                // CoalesceBatchesExec combines small batches into larger batches for more efficient writing
+                let physical_input = Arc::new(CoalesceBatchesExec::new(
+                    physical_inputs[0].clone(),
+                    batch_size,
+                ));
```

Comment:
This does not need to be added, the actual write operation is `TableWriterPlanNode `, and `TableWriterMergePlanNode ` just summarizes the write results